lib/routing/sfRoute.class.php

```diff
@@ -759,7 +764,7 @@ protected function generateStarParameter($url, $defaults, $parameters)
       }
       else
       {
-        $tmp[] = urlencode($key).'/'.urlencode($value);
+        $tmp[] = urlencode($key) . '/' . urlencode($value ?: '');
```

Comment:
![arrested-development-season1](https://user-images.githubusercontent.com/69626/212793764-2f6e2e3e-7607-46b4-9fa9-215a21cf0960.gif)

This tiny change breaks any URL where parameters use `1` and `0` to indicate boolean flags. For example, the URL
> https://manager-dev.ah.dd81.controlport.co.uk/trolleys/active/0/status/0/type/Trolley

gets turned into

> https://manager-dev.ah.dd81.controlport.co.uk/trolleys/active//status//type/Trolley

A patch to fix will be forthcoming.